goversion/goversion.go

```diff
@@ -156,3 +156,44 @@ func extractPrerelease(part, prerelease *string) {
 		*part = (*part)[:i]
 	}
 }
+
+type GoVersions []*GoVersion
```

Comment:
My concern with this and the old Less method is mainly that we're providing "weird" behavior in an exported API. But thinking about it again, we can just make it clear what "breaks" it.

```suggestion
// GoVersions implements [sort.Interface] and sorts versions in descending order.
// If Major, Minor, Patch, or Revision of any GoVersion in the slice can't be parsed by
// [strconv.Atoi], the result of using this type is undefined.
type GoVersions []*GoVersion
```